src/Npgsql/Internal/NpgsqlConnector.Auth.cs

```diff
@@ -19,35 +19,41 @@ partial class NpgsqlConnector
 {
     async Task Authenticate(string username, NpgsqlTimeout timeout, bool async, CancellationToken cancellationToken)
     {
-        timeout.CheckAndApply(this);
-        var msg = ExpectAny<AuthenticationRequestMessage>(await ReadMessage(async), this);
-        switch (msg.AuthRequestType)
+        var authenticated = false;
+        while (!authenticated)
         {
-        case AuthenticationRequestType.AuthenticationOk:
-            return;
+            timeout.CheckAndApply(this);
+            var msg = ExpectAny<AuthenticationRequestMessage>(await ReadMessage(async), this);
+            switch (msg.AuthRequestType)
+            {
+            case AuthenticationRequestType.AuthenticationOk:
+                authenticated = true;
```

Comment:
nit: can just return directly from here, and do a `while (true)` above instead

Reply:
@roji 
Thank you, it's now fixed.